atomic_reactor/plugins/gather_builds_metadata.py

```diff
@@ -54,6 +57,56 @@ def _generate_build_log_output(self, platform: str, buildroot_id: int) -> Option
         metadata['arch'] = platform
         return Output(metadata=metadata, filename=str(build_log_file))
 
+    def _get_hostname_for_platform(self, platform: str) -> str:
+        task_results = self.workflow.osbs.get_task_results(self.workflow.pipeline_run_name)
+        task_platform = platform.replace('_', '-')
+        build_host = ""
+
+        for task_name in task_results:
+            if not (task_name.startswith('binary-container-build') and task_platform in task_name):
+                continue
+
+            if 'task_result' not in task_results[task_name]:
+                raise RuntimeError(f"task_results is missing from: {task_name}")
+
+            build_host = json.loads(task_results[task_name]['task_result'])
+            break
```

Comment:
even here we have a for loop, continue and break statements when looking for one element, it does not seem readable or pythonic. I would prefer something like:
```
task_name = next(filter(lambda task_name: task_name.startswith('binary-container-build'), task_results))
...rest of the code
```
It seems more readable, less lines and it has straightforward flow, so you don't have to watch the continue, break statements.

Reply:
ok